src/domain/safe/entities/transfer.entity.ts

```diff
@@ -12,11 +11,43 @@ export interface ERC20Transfer extends Transfer {
   value: string;
 }
 
+export function isERC20Transfer(transfer: Transfer): transfer is ERC20Transfer {
+  const erc20Transfer = transfer as ERC20Transfer;
+  return (
+    erc20Transfer.tokenAddress !== undefined &&
+    erc20Transfer.tokenAddress !== null &&
+    erc20Transfer.value !== undefined &&
+    erc20Transfer.value !== null
+  );
+}
+
 export interface ERC721Transfer extends Transfer {
   tokenAddress: string | null;
   tokenId: string;
 }
 
+export function isERC721Transfer(
+  transfer: Transfer,
+): transfer is ERC721Transfer {
+  const erc721Transfer = transfer as ERC721Transfer;
+  return (
+    erc721Transfer.tokenAddress !== undefined &&
+    erc721Transfer.tokenAddress !== null &&
+    erc721Transfer.tokenId !== undefined &&
+    erc721Transfer.tokenId !== null
+  );
+}
+
 export interface NativeTokenTransfer extends Transfer {
   value: string;
 }
+
+export function isNativeTokenTransfer(
+  transfer: Transfer,
+): transfer is NativeTokenTransfer {
+  const nativeTokenTransfer = transfer as NativeTokenTransfer;
+  return (
+    nativeTokenTransfer.value !== undefined &&
+    nativeTokenTransfer.value !== null
+  );
```

Comment:
As you mentioned the following payload:

```typescript
{
  value: 123,
  tokenAddress: "0x00"
}
```

can be considered both a `NativeTokenTransfer` and a `ERC20Transfer`. My suggestion is to rewrite it as:

```typescript
export function isNativeTokenTransfer(
  transfer: Transfer,
): transfer is NativeTokenTransfer {
  const nativeTokenTransfer = transfer as NativeTokenTransfer;
  return (
    (transfer as any).tokenAddress == null && nativeTokenTransfer.value != null
  );
}
```

Therefore we only consider a transfer to be a `NativeTokenTransfer` IFF `tokenAddress` is `null` or `undefined` AND `value` is not `null` or `undefined`.

Wdyt?

Reply:
Yes, I think that's a more solid definition. Thanks, changed in [a41f860](https://github.com/5afe/safe-client-gateway-nest/pull/206/commits/a41f8600dc9fbe237de992a74c51dce6a6703668)